ompi/mca/op/rocm/op_rocm_functions.c

```diff
@@ -0,0 +1,1897 @@
+/*
+ * Copyright (c) 2019-2023 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2020      Research Organization for Information Science
+ *                         and Technology (RIST).  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+#ifdef HAVE_SYS_TYPES_H
+#include <sys/types.h>
+#endif
+#include "opal/util/output.h"
+
+
+#include "ompi/op/op.h"
+#include "ompi/mca/op/op.h"
+#include "ompi/mca/op/base/base.h"
+#include "ompi/mca/op/rocm/op_rocm.h"
+#include "opal/mca/accelerator/accelerator.h"
+
+#include "ompi/mca/op/rocm/op_rocm.h"
+#include "ompi/mca/op/rocm/op_rocm_impl.h"
+
+/**
+ * Disable warning about empty macro var-args.
+ * We use varargs to suppress expansion of typenames
+ * (e.g., int32_t -> int) which could lead to collisions
+ * for similar base types. */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
+
+static inline void device_op_pre(const void *orig_source1,
```

Comment:
@bosilca @edgargabriel 
If the `device_op_pre` and `device_op_post` use the accelerator framework they are pretty much independent of the the model (minus the last two lines). I wonder whether they should be moved to a header in `base/` and shared between the two implementations. 

The last two lines can be taken out and put into the op macro from where they are called.